src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -220,8 +220,11 @@ public void StartConnectAsync()
             }
             catch (OperationCanceledException timeout)
             {
-                this.logger.LogDebug("Peer {0} connection timeout.", peerAddress.NetworkAddress.Endpoint);
-                peer?.DisconnectWithException("Timeout", timeout);
+                this.logger.LogDebug(this.nodeLifetime.ApplicationStopping.IsCancellationRequested? "Peer {0} connection canceled because application is stopping." :
```

Comment:
Can we maybe rewrite this so that we evaluate `this.nodeLifetime.ApplicationStopping.IsCancellationRequested` once? i.e.

```
if (this.nodeLifetime.ApplicationStopping.IsCancellationRequested)
{    
    this.logger.LogDebug("Peer {0} connection canceled because application is stopping.", peerAddress.NetworkAddress.Endpoint);
    peer?.DisconnectWithException("Application stopping");
}
else
{
    this.logger.LogDebug("Peer {0} connection timeout.", peerAddress.NetworkAddress.Endpoint);
    peer?.DisconnectWithException("Timeout");
}  
```

Reply:
Sure.
Fixed.